src/frontend/sharedComponents/LeaveProjectModalContent/LeaveProject.tsx

```diff
@@ -0,0 +1,193 @@
+import {BottomSheetModalContent} from '../BottomSheetModal';
+import {useAcceptInvite} from '../../hooks/server/invites';
+import {ErrorBottomSheet} from '../ErrorBottomSheet';
+
+const m = defineMessages({
+  leaveProj: {
+    id: 'screens.LeaveProject.leaveProj',
+    defaultMessage: 'Leave Project',
+  },
+  deleteConsentWithName: {
+    id: 'screens.LeaveProject.deleteConsentWithName',
+    defaultMessage:
+      'I understand I will be deleting all data from Project {projectName} from my device.',
+  },
+  deleteConsentWithoutName: {
+    id: 'screens.LeaveProject.deleteConsentWithoutName',
+    defaultMessage: 'I understand I will be deleting all data from my device.',
+  },
+  removeFromProjWithName: {
+    id: 'screens.LeaveProject.removeFromProjWithName',
+    defaultMessage:
+      "This will remove all Project {projectName}'s data from your device.",
+  },
+  removeFromProjWithoutName: {
+    id: 'screens.LeaveProject.removeFromProjWithoutName',
+    defaultMessage: 'This will remove all of the data from your device.',
+  },
+  cancel: {
+    id: 'screens.LeaveProject.cancel',
+    defaultMessage: 'Cancel',
+  },
+  checkToConfirm: {
+    id: 'screens.LeaveProject.checkToConfirm',
+    defaultMessage: 'Please check the box to confirm',
+  },
+  leavingProject: {
+    id: 'screens.LeaveProject.leavingProject',
+    defaultMessage: 'Leaving Project {projectName}',
+  },
+});
+
+type LeaveProjectProps = {
+  inviteId: string;
+  accept: ReturnType<typeof useAcceptInvite>;
+
+  closeSheet: () => void;
+  projectName?: string;
+};
+
+export const LeaveProject = ({
+  inviteId,
+  accept,
+  closeSheet,
+  projectName,
+}: LeaveProjectProps) => {
+  const {formatMessage} = useIntl();
+  const [error, setError] = React.useState(false);
+  const [isChecked, setIsChecked] = React.useState(false);
+  const leaveProject = useLeaveProject();
+  const [combinedLoading, setCombinedLoading] = React.useState(false);
+
+  function handleLeavePress() {
+    if (!isChecked) {
+      setError(true);
+      return;
+    }
+    setCombinedLoading(true);
+    // we want to accept first because the invitor will be able to cancel. this avoids the user leaving a project, and then their invite being cancelled before they were able to join.
+    accept.mutate(
+      {inviteId},
+      {
+        onSuccess: () => {
+          closeSheet();
+          setCombinedLoading(false);
+        },
+        // This is commented out for now and issue created: https://github.com/digidem/comapeo-mobile/issues/525
+        // onSuccess: () => {
+        //   leaveProject.mutate(undefined, {
+        //     onSuccess: () => {
+        //       closeSheet();
+        //       setCombinedLoading(false);
+        //     },
+        //     onError: err => {
+        //       console.log(err);
+        //       setCombinedLoading(false);
+        //     },
+        //   });
+        // },
+        onError: err => {
+          console.log(err);
```

Comment:
nit: remove or replace with sentry?

```suggestion
          console.log(err);
```

Reply:
https://github.com/digidem/comapeo-mobile/pull/526/commits/75b65e12cfa63aa835b3e81d369d653ef7d6b3e9